words/headers.asm

```diff
@@ -51,1446 +58,460 @@
+#nt_header get_order, "get-order"
+#nt_header set_order, "set-order"
+#nt_header get_current, "get-current"
+#nt_header set_current, "set-current"
+#nt_header search_wordlist, "search-wordlist"
+#nt_header wordlist
+#nt_header definitions
 .endif
 
-.if "editor" in TALI_OPTIONAL_WORDS && "wordlist" in TALI_OPTIONAL_WORDS
-nt_editor_wordlist:     ; shares code with ONE
-        .byte 15, 0
-        .word +, xt_editor_wordlist, z_editor_wordlist
-        .text "editor-wordlist"
-+
-.endif
+.if "block" in TALI_OPTIONAL_WORDS
 
-.if "wordlist" in TALI_OPTIONAL_WORDS
-nt_forth_wordlist:      ; shares code with ZERO
-        .byte 14, 0
-        .word nt_only, xt_forth_wordlist, z_forth_wordlist
-        .text "forth-wordlist"
-
-nt_only:
-        .byte 4, 0
-        .word nt_also, xt_only, z_only
-        .text "only"
-
-nt_also:
-        .byte 4, 0
-        .word nt_previous, xt_also, z_also
-        .text "also"
-
-nt_previous:
-        .byte 8, 0
-        .word nt_to_order, xt_previous, z_previous
-        .text "previous"
-
-nt_to_order:
-        .byte 6, 0
-        .word nt_order, xt_to_order, z_to_order
-        .text ">order"
-
-nt_order:
-        .byte 5, 0
-        .word nt_forth, xt_order, z_order
-        .text "order"
-
-nt_forth:
-        .byte 5, 0
-        .word +, xt_forth, z_forth
-        .text "forth"
-+
+    .if "ramdrive" in TALI_OPTIONAL_WORDS
+#nt_header block_ramdrive_init, "block-ramdrive-init"
+    .endif
+   .if TALI_ARCH == "c65"
+#nt_header block_c65_init, "block-c65-init"
+   .endif
+   .if "editor" in TALI_OPTIONAL_WORDS
+#nt_header list
+   .endif
+
+#nt_header thru
+#nt_header load
+#nt_header flush
+#nt_header empty_buffers, "empty-buffers"
+#nt_header buffer
+#nt_header update
+#nt_header block
+#nt_header save_buffers, "save-buffers"
+#nt_header block_read_vector, "block-read-vector", NN
+#nt_header block_read, "block-read", NN
+#nt_header block_write_vector, "block-write-vector", NN
+#nt_header block_write, "block-write", NN
+#nt_header blk, "blk", NN
+#nt_header scr, "scr", NN
+#nt_header blkbuffer
+#nt_header buffblocknum
+#nt_header buffstatus
 .endif
 
-nt_see: .byte 3, NN
-        .word +, xt_see, z_see
-        .text "see"
-+
-
-.if "ed" in TALI_OPTIONAL_WORDS
-nt_ed:                  ; ed6502
-        .byte 3, NN
-        .word +, xt_ed, z_ed
-        .text "ed:"
-+
+.if "environment?" in TALI_OPTIONAL_WORDS
+#nt_header environment_q, "environment?"
 .endif
 
-nt_cold:
-        .byte 4, 0
-        .word nt_bye, xt_cold, z_cold
-        .text "cold"
+.if "disassembler" in TALI_OPTIONAL_WORDS
+#nt_header disasm
+.endif
 
-nt_bye:
-        .byte 3         ; length of word strings
-        .byte 0         ; status byte
-        .word 0000      ; next word in Dictionary, 0000 signals end
-        .word xt_bye    ; start of code block (xt of this word)
-        .word z_bye     ; end of code (RTS)
-        .text "bye"     ; word name, always lower case, not zero-terminated
+#nt_header see, "see", NN
```

Comment:
Note the header word order is reversed since the macro is linking each NT to the previous one in memory (mimicking the way user words are allocated in the dictionary) instead of the following one in memory.  This puts the most common words towards the end of this file.

In principle we could save a bunch of extra space by putting each each definition just before the code body but that makes it harder to put common words near the beginning of the search order without breaking the mostly alphabetical order of implementation and the split across word files.

At least for the main wordlist I'd leave all the headers here.  For the other wordlists I'd lean towards moving their headers into the implementation files near the word bodies but I didn't want to make extra changes in this already large PR...